### DISCUSSION TITLE
Question regarding NATIVE6 and ERA5 variables mapping

### DISCUSSION THREAD

**Discussion by rbeucher** [ESMValGroup/ESMValCore#2493] [General]
Hi all,

This question is directed primarily at the @ESMValGroup/technical-lead-development-team, but it may also be beneficial for the broader community.

Regarding the Native6 ERA5 fixes, could you please explain how the mapping from ERA5 variable names to CMIP names is handled? I understand that the data is organized in the Tier3 folder on Levante, using the following directory structure: Tier{tier}/{dataset}/{version}/{frequency}/{short_name}, where short_name refers to the CMIP variable short name.

The reason I ask is that we are currently using a similar directory structure with symlinks pointing to the actual NetCDF files stored in the ERA5 national collection at NCI. This setup introduces some maintenance overhead, as the links need to be updated when new data is added. I'm exploring ways we could directly use the ERA5 collections to streamline this process.

Any insights or suggestions would be greatly appreciated.

Romain

**Comment by bouweandela** [comment]
> Regarding the Native6 ERA5 fixes, could you please explain how the mapping from ERA5 variable names to CMIP names is handled? I understand that the data is organized in the Tier3 folder on Levante, using the following directory structure: Tier{tier}/{dataset}/{version}/{frequency}/{short_name}, where short_name refers to the CMIP variable short name.

There isn't really any mapping or safety mechanism in place, any files found in the folder *rootpath*/Tier{tier}/{dataset}/{version}/{frequency}/{short_name} are considered to represent the correct ERA5 variable. It is up to the person downloading the data to put the files with the right variable in the right directory. We have fairly extensive documentation available on how the `rootpath` and `drs` settings and config-developer.yml work [here](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/quickstart/find_data.html) and [here](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/quickstart/configure.html#developer-configuration-file).

To directly use the ERA5 national collection at NCI you could add a an extra entry here
https://github.com/ESMValGroup/ESMValCore/blob/4c21ca64185cc253ac896ee13c623afd168d7bc3/esmvalcore/config-developer.yml#L100-L101
and if you need extra facets that are automatically added you can add those as documented [here](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/quickstart/configure.html#extra-facets).

To see an example of how this could be done, you can have a look at #2178.
